Introduce Towncrier to manage the release notes

abompard opened the pull request:
This should help a lot with the next releases.

There are changes in the developer guidelines, please review them and create the files in the `news` directory accordingly.
Also, when reviewing PRs, please make sure that the submitter did not forget to add a news fragment and a migration fragment when appropriate, as well as their author fragment, in the `news` directory.

Fixes #3562

abompard committed:
Introduce Towncrier to manage the release notes

Fixes #3562

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

Zlopez commented on news/get-authors.py:
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
```
I recommend to add some description of this file. You could copy it from Anitya - https://github.com/release-monitoring/anitya/blob/master/news/get-authors.py

abompard replied:
Added, thanks.

Zlopez commented on docs/developer/releases.rst:
```diff
@@ -50,29 +50,23 @@ How to make a release
-   ``3.13``, you could use ``git log 3.12..3.13``. Be aware that any commits that were cherry picked
-   to ``3.12`` will show up in this example and we wouldn't want to mark those commits as being new
-   in ``3.13``.
+If you are making a new major or minor release:
```
Does bodhi uses separate package for fedora messaging schemes with separate versioning?

If yes, there should be also mentioned here.

abompard replied:
No, all sub-packages have the same version, including the message schemas.

Zlopez commented on docs/developer/index.rst:
```diff
@@ -82,6 +80,39 @@ Before you submit a pull request to Bodhi, please ensure that it meets these cri
+Release Notes
+-------------
+
+To add entries to the release notes, create a file in the ``news`` directory in the
```
I would recommend to switch `type` and `source` order, because I always need to check if it's `source.type` or `type.source`. This is caused by type being explained first and then source comes to play.

abompard replied:
Yeah that's more readable, I used to use the other order because I wanted to introduce the author type first because it was a special case in the source part, but now I've reordered the docs and that's no longer necessary. I'll switch them.

Zlopez commented on docs/developer/index.rst:
```diff
@@ -82,6 +80,39 @@ Before you submit a pull request to Bodhi, please ensure that it meets these cri
+* If your changes contain database migrations, you must add a file named ``summary.migration``
+  in the ``news`` directory, where ``summary`` is a short textual description of the change
+  (or the issue number you're fixing).
+* If it is not present already, add a file in the ``news`` directory named ``username.author``
```
I don't think this is needed if you have `get-authors.py` script.

abompard replied:
Yeah, that was more in case the get-author script makes mistakes, which I've seen it do because some authors use weird or inconsistent email addresses in their git config (like `someone+github@example.com`), so I wanted to offer a way out.

Also, I think it's more and information that's part of the patch than part of the release process commit. I see the get-authors script as a safety net to avoid forgetting someone, not as the plan A for doing things.

Zlopez reviewed (commented):
Just few comments, otherwise looks :+1:

abompard committed:
Improve documentation

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

abompard committed:
Fix RST syntax

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

abompard committed:
Make it possible to archive the docs produced by bodhi-ci

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

abompard committed:
Remove obsolete MANIFEST declaration

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

abompard committed:
Prevent pip from using its PEP517 backend

We're doing wonky stuff in our setup.py and it falls apart if pip uses
PEP-517, which is triggered by the presence of the pyproject.toml file.

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

abompard committed:
Ignore the test results directory mounted by bodhi-ci

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>
